test/support/version.rb

```diff
@@ -0,0 +1,55 @@
+class SpecVersion
+  # If beginning implementations have a problem with this include, we can
+  # manually implement the relational operators that are needed.
+  include Comparable
+
+  # SpecVersion handles comparison correctly for the context by filling in
+  # missing version parts according to the value of +ceil+. If +ceil+ is
+  # +false+, 0 digits fill in missing version parts. If +ceil+ is +true+, 9
+  # digits fill in missing parts. (See e.g. VersionGuard and BugGuard.)
+  def initialize(version, ceil = false)
+    @version = version
+    @ceil    = ceil
+    @integer = nil
+  end
+
+  def to_s
+    @version
+  end
+
+  def to_str
+    to_s
+  end
+
+  # Converts a string representation of a version major.minor.tiny
+  # to an integer representation so that comparisons can be made. For example,
+  # "2.2.10" < "2.2.2" would be false if compared as strings.
+  def to_i
+    unless @integer
+      major, minor, tiny = @version.split "."
+      if @ceil
+        tiny = 99 unless tiny
+      end
+      parts = [major, minor, tiny].map do |x|
+        x = x.to_i
+        x < 10 ? "0#{x}" : x.to_s
+      end
+      @integer = "1#{parts.join}".to_i
+    end
+    @integer
+  end
+
+  def to_int
+    to_i
+  end
+
+  def <=>(other)
+    if other.respond_to? :to_int
+      other = other.to_int
+    else
+      other = SpecVersion.new(other.to_s).to_i
+    end
+
+    self.to_i <=> other
+  end
+end
```

Comment:
Since this is a direct copy of the file from the mspec library, we will need to make a note in the README about using part of mspec, which is MIT. Up until now, we had re-implemented everything that we needed, not directly looking at mspec code. This is the first file where it was just copied.

Reply:
Oh, and we should add the copyright to the top of the file just to be clear we didn't write this...